aot-maven-plugin/src/main/java/com/dylibso/chicory/experimental/maven/aot/AotGenMojo.java

```diff
@@ -123,12 +129,126 @@ public void execute() throws MojoExecutionException {
+        var methodBody = method.createBody();
+
+        var constructorInvocation =
+                new ObjectCreationExpr(
+                        null,
+                        parseClassOrInterfaceType(machineName),
+                        NodeList.nodeList(new NameExpr("instance")));
+        methodBody.addStatement(new ReturnStmt(constructorInvocation));
+
+        dest.add(packageName, machineFactoryName + ".java", cu);
+    }
+
+    private static void generateModule(
+            SourceRoot dest, String packageName, String moduleName, String wasmName) {
+        var cu = new CompilationUnit(packageName);
+
+        cu.addImport(IOException.class);
+        cu.addImport(UncheckedIOException.class);
+        cu.addImport(Parser.class);
+        cu.addImport(WasmModule.class);
+
+        var clazz = cu.addClass(moduleName, Keyword.PUBLIC, Keyword.FINAL);
+
+        var constr = clazz.addConstructor(Keyword.PRIVATE);
+        constr.createBody();
+
+        var method = clazz.addMethod("load", Keyword.PUBLIC, Keyword.STATIC);
+        method.setType(WasmModule.class);
+        var methodBody = method.createBody();
+
+        var getResource =
+                new MethodCallExpr(
+                        new ClassExpr(parseType(moduleName)),
+                        "getResourceAsStream",
+                        new NodeList<>(new StringLiteralExpr(wasmName)));
+        var resourceVar =
+                new VariableDeclarationExpr(
+                        new VariableDeclarator(new VarType(), "in", getResource));
+
+        var returnStmt =
+                new ReturnStmt(
+                        new MethodCallExpr()
+                                .setScope(new NameExpr("Parser"))
+                                .setName("parse")
+                                .addArgument(new NameExpr("in")));
+
+        var newException =
+                new ObjectCreationExpr()
+                        .setType(parseClassOrInterfaceType("UncheckedIOException"))
+                        .addArgument(new StringLiteralExpr("Failed to load AOT WASM module"))
+                        .addArgument(new NameExpr("e"));
+        var catchIoException =
+                new CatchClause()
+                        .setParameter(
+                                new com.github.javaparser.ast.body.Parameter(
+                                        parseClassOrInterfaceType("IOException"), "e"))
+                        .setBody(new BlockStmt(new NodeList<>(new ThrowStmt(newException))));
+
+        methodBody.addStatement(
+                new TryStmt()
+                        .setResources(new NodeList<>(resourceVar))
+                        .setTryBlock(new BlockStmt(new NodeList<>(returnStmt)))
+                        .setCatchClauses(new NodeList<>(catchIoException)));
+
+        dest.add(packageName, moduleName + ".java", cu);
+    }
+
+    private static byte[] rewriteWasm(byte[] wasmBytes, WasmModule module) {
```

Comment:
Instead of rewriting in-place the existing wasm file, we can place the original `.wasm` files outside `resources` and produce the result in `target/generated-resources`.

Reply:
already doing it